src/client/lib/consentsTracking.ts

```diff
@@ -0,0 +1,79 @@
+import { PageData } from '@/shared/model/GlobalState';
+import { Routes } from '@/shared/model/Routes';
+import { sendOphanInteractionEvent } from './ophan';
+
+const trackInputElementInteraction = (
+  inputElem: HTMLInputElement,
+  component: string,
+  consentName: string,
+) => {
+  if (inputElem.type === 'checkbox') {
+    sendOphanInteractionEvent({
+      component: `identity-onboarding-${component}`,
+      value: `${consentName} : ${inputElem.checked}`,
+    });
+  }
+
+  if (inputElem.type === 'radio' && inputElem.checked) {
+    sendOphanInteractionEvent({
+      component: `identity-onboarding-${component}`,
+      value: `${consentName} : ${inputElem.value}`,
+    });
+  }
+};
+
+// handle consents form submit event
+const consentsFormSubmitOphanTracking = (
+  inputElems: NodeListOf<HTMLInputElement>,
+  pageData: PageData,
+) => {
+  const consents = pageData.consents;
+
+  if (consents) {
+    inputElems.forEach((elem) => {
+      const consent = consents.find((c) => c.id === elem.name);
+
+      if (consent) {
+        trackInputElementInteraction(elem, 'consent', consent.name);
+      }
+    });
+  }
+};
+
+// handle newsletter form submit event
+const newslettersFormSubmitOphanTracking = (
+  inputElems: NodeListOf<HTMLInputElement>,
+  pageData: PageData,
+) => {
+  const newsletters = pageData.newsletters;
+
+  if (newsletters) {
+    inputElems.forEach((elem) => {
+      const newsletter = newsletters.find((c) => c.id === elem.name);
```

Comment:
could you destructure here ?  `find ( ({id}) => id === elem.name` 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find#Using_arrow_function_and_destructuring

Reply:
Great idea!